server/routes/v1/tests/user.test.ts

```diff
@@ -0,0 +1,97 @@
+import { Request, Response } from "express";
+
+import db from "../../../models/index";
+import { User } from "../../../models/user";
+import { signUpHandler, signInHandler } from "../user";
+
+function mockRequest(data: any) {
+  return { body: data };
+}
+
+function mockResponse() {
+  const res: any = {};
+  res.status = jest.fn().mockReturnValue(res);
+  res.json = jest.fn().mockReturnValue(res);
+  res.cookie = jest.fn().mockReturnValue(res);
+  return res;
+}
+
+beforeAll(async () => {
+  await db.sequelize.sync({ force: true });
+});
+
+afterAll(async () => {
+  await db.sequelize.sync({ force: true });
+});
+
+describe("v1 - User Routes", () => {
+  describe("/signup endpoint", () => {
+    test("If successful signup, returns 201", async () => {
+      const data = {
+        email: "email@mail.utoronto.ca",
+        userName: "userName",
+        password: "password",
+        firstName: "firstName",
+        lastName: "lastName",
+      };
+      const req = mockRequest(data) as Request;
+      const res = mockResponse() as Response;
+      await signUpHandler(req, res);
+      expect(res.status).toHaveBeenCalledWith(201);
+      expect(res.json).toHaveBeenCalled();
+    });
+
+    test("If unsuccessful signup, returns error status", async () => {
+      const data = {
+        email: "email@mail.yahoo.ca",
+        userName: "userName",
+        password: "password",
+        firstName: "firstName",
+        lastName: "lastName",
+      };
+      const req = mockRequest(data) as Request;
+      const res = mockResponse() as Response;
+      await signUpHandler(req, res);
+      expect(res.status).not.toHaveBeenCalledWith(201);
```

Comment:
Can we assert a specific code? This is brittle since calling with 200, 204 will all pass